roofit/roofitcore/src/RooNLLVar.cxx

```diff
@@ -500,11 +500,18 @@ std::tuple<double, double, double> RooNLLVar::computeBatched(std::size_t stepSiz
     _dataClone->get(evtNo);
     assert(_dataClone->valid());
     try {
-      BatchHelpers::BatchInterfaceAccessor::checkBatchComputation(*pdfClone, *_evalData, evtNo-firstEvent, _normSet);
+      // Cross check results with strict tolerance and complain
+      BatchHelpers::BatchInterfaceAccessor::checkBatchComputation(*pdfClone, *_evalData, evtNo-firstEvent, _normSet, 1.E-13);
     } catch (std::exception& e) {
-      std::cerr << "ERROR when checking batch computation for event " << evtNo << ":\n"
+      std::cerr << __FILE__ << ":" << __LINE__ << " ERROR when checking batch computation for event " << evtNo << ":\n"
           << e.what() << std::endl;
-      assert(false);
+
+      // It becomes a real problem if it's very wrong. We fail in this case:
+      try {
+        BatchHelpers::BatchInterfaceAccessor::checkBatchComputation(*pdfClone, *_evalData, evtNo-firstEvent, _normSet, 1.E-9);
+      } catch (std::exception& e2) {
+        assert(false);
```

Comment:
Don't want to use here an assert that is there also for non-debug builds?
Something like R__ASSERT ? Or ROOT::Fatal() ?

Reply:
I didn't see the need. It's error checking code that only runs when you compile in debug mode. It's something like 3x slower, so we cannot run it in production.

Reply:
ok